### PR TITLE
framework/st_things: Add function to debug payload for request/response.

### PR DESCRIPTION
**Pull request by Hyunjun85**
It can show the payload what attributes are included
when enabled the CONFIG_DEBUG_ST_THINGS_DEBUG config.

log format example:
IncomingRequest: (id: 0)
- resourceHandle: 0x00000000
- uri: "/capability/switch/main/0"
- property: 0x0
- flag: 0x0
			  REQUEST
 			  OBSERVE
- method: 1
 			  GET
- query: 'if=oic.if.baseline'
	- values:
 	  {"power":  "off"}
- dev_addr: adapter:00, flags:0x00, port:000

OutgoingResponse:
   - resourceHandle: 0x0000000
   - result: 0
   - resourceUri: ''
   - persistentBufferFlag: 0
   - payload-type: 4 (Representation)
 	- uri: /capability/switch/main/0
 	- types:
		  x.com.st.powerswitch
		  x.com.st.powerswitch
	- interfaces:
		  oic.if.baseline
		  oic.if.a
		  oic.if.baseline
		  oic.if.a
	- values:
 	  {"power":  "off"}

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [ec9f2f9fc4c87f096836f121f7cf6b58de193ea3] - Code Rule Check Result:
framework/src/st_things/things_stack/logging/things_logger.c:351: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('

**Comment by seinfra** [comment]
Target : [ec9f2f9fc4c87f096836f121f7cf6b58de193ea3] - Code Rule Check (C++) OK.

File: framework/src/st_things/things_stack/logging/things_logger.c [hunk 1]
```diff
@@ -130,3 +133,252 @@ void things_logv(things_log_level_e level, const char *tag, const char *func_nam
...
+		default:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  unknown} ");
+			break;
+		}
+
+		cur = cur->next;
+	}
+}
+
+static void things_log_payload_flag(things_log_level_e level, const char *tag, const char *func_name, const int16_t line_num,
```
Comment (KimDongEon) [inline framework/src/st_things/things_stack/logging/things_logger.c]: What is the difference between a function with "_" under bar and a function without function name?

_coap_opt_name ()
_ll ()
_ll_rep_value ()

Would not it be a good idea to prefix the under bar with static functions like this?

Comment (Hyunjun85) [reply framework/src/st_things/things_stack/logging/things_logger.c]: Well.. My thingking those function will use only in this file spaces. So, I copied function name from function name of slsi debug function's name. 
However, It looks not good.

**Comment by seinfra** [comment]
Target : [dcb9f358f73823cc413d9c426d54d83db2c392b6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [dcb9f358f73823cc413d9c426d54d83db2c392b6] - Code Rule Check OK.

File: framework/src/st_things/things_stack/logging/things_logger.c [hunk 2]
```diff
@@ -130,3 +133,252 @@ void things_logv(things_log_level_e level, const char *tag, const char *func_nam
...
+	{ 140, "Reserved" },
+	{ 258, "No-Response" },
+	{ 2049, "OCF-Accept-Content-Format-Version" },
+	{ 2053, "OCF-Content-Format-Version" },
+};
+
+static const char *_coap_opt_name(int number)
+{
+	unsigned int i;
+
```
Comment (thapav) [inline framework/src/st_things/things_stack/logging/things_logger.c]: please add input parameter check

Comment (Hyunjun85) [reply framework/src/st_things/things_stack/logging/things_logger.c]: Thanks for your comment.
Applied.

**Comment by seinfra** [comment]
Target : [d764396169cb36f0e40e50e09c9ebdba69acbffa] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d764396169cb36f0e40e50e09c9ebdba69acbffa] - Code Rule Check OK.

**Pull request merged**